python_onnxifi/data_conversion.cc

```diff
@@ -0,0 +1,251 @@
+  } 	                   							 \
+  case ONNX_NAMESPACE::TensorProto_DataType_COMPLEX64: {			 \
+    op_template<std::complex<float>, std::complex<float>>(__VA_ARGS__);          \
+    break;									 \
+  }										 \
+  case ONNX_NAMESPACE::TensorProto_DataType_FLOAT16:  { 			 \
+   /*TODO:op_template<int32_t, half>(__VA_ARGS__);*/                       	 \
+    break;									 \
+  }				 						 \
+  case ONNX_NAMESPACE::TensorProto_DataType_BOOL:  { 				 \
+    op_template<int32_t, bool>(__VA_ARGS__);          				 \
+    break;									 \
+  } 										 \
+  case ONNX_NAMESPACE::TensorProto_DataType_INT8:  {				 \
+    op_template<int32_t, int8_t>(__VA_ARGS__);                     		 \
+    break;									 \
+  } 	   	      								 \
+  case ONNX_NAMESPACE::TensorProto_DataType_INT16:  {				 \
+    op_template<int32_t, int16_t>(__VA_ARGS__);          			 \
+    break;									 \
+  } 										 \
+  case ONNX_NAMESPACE::TensorProto_DataType_INT32:  { 				 \
+    op_template<int32_t, int32_t>(__VA_ARGS__);          			 \
+    break;									 \
+  } 			       							 \
+  case ONNX_NAMESPACE::TensorProto_DataType_UINT8:  {				 \
+    op_template<int32_t, uint8_t>(__VA_ARGS__);          			 \
+    break;									 \
+  } 		 								 \
+  case ONNX_NAMESPACE::TensorProto_DataType_UINT16: { 				 \
+    op_template<int32_t, uint16_t>(__VA_ARGS__);          		    	 \
+    break;									 \
+  } 									         \
+  case ONNX_NAMESPACE::TensorProto_DataType_INT64: {				 \
+    op_template<int64_t, int64_t>(__VA_ARGS__);                               	 \
+    break;                                                               	 \
+  }										 \
+  case ONNX_NAMESPACE::TensorProto_DataType_UINT32:  {				 \
+    op_template<uint64_t, uint32_t>(__VA_ARGS__);                                \
+    break;                                                                       \
+  }     									 \
+  case ONNX_NAMESPACE::TensorProto_DataType_UINT64: {			 	 \
+    op_template<uint64_t, uint64_t>( __VA_ARGS__);                            	 \
+    break;							        	 \
+  }   						                        	 \
+  case ONNX_NAMESPACE::TensorProto_DataType_DOUBLE: {                   	 \
+   op_template<double, double>(__VA_ARGS__);                                 	 \
+   break;                                                                        \
+  }										 \
+  case ONNX_NAMESPACE::TensorProto_DataType_COMPLEX128: {     			 \
+   op_template<std::complex<double>, std::complex<double>>(__VA_ARGS__);         \
+   break;									 \
+  }										 \
+  case ONNX_NAMESPACE::TensorProto_DataType_STRING: 				 \
+  case ONNX_NAMESPACE::TensorProto_DataType_UNDEFINED: {			 \
+   throw std::runtime_error("Dispatch received non-numeric data type");          \
+  }										 \
+}										 \
+
+template<typename onnx_type, typename unused>
+void DataConversion::getBufferSize(uint64_t& buffer_size, const uint64_t* shape, uint32_t dimensions)  {
```

Comment:
This is a weird interface. Why not 
```
unint64_t DataConversion::getBufferSize(uint64_t shape, uint32_t dimension)
```

Reply:
Since the template is dispatched (part of switch statement that executes templated void function), I think this should be void as well. Otherwise, I think it complicates the dispatch macro.

Reply:
Ah, I missed that. Cool!